modules/integration/src/test/scala/scala/cli/integration/FixTests.scala

```diff
@@ -0,0 +1,319 @@
+      val projectFileContents = os.read(root / projectFileName)
+      val mainFileContents    = os.read(root / mainFileName)
+
+      assertNoDiff(
+        projectFileContents,
+        """// Main
+          |//> using objectWrapper
+          |
+          |//> using dependency "com.lihaoyi::os-lib:0.9.1"
+          |//> using dependency "com.lihaoyi::pprint:0.6.6"
+          |//> using dependency "com.lihaoyi::upickle:3.1.2"
+          |
+          |""".stripMargin
+      )
+
+      assertNoDiff(
+        mainFileContents,
+        """package com.foo.main
+          |
+          |object Main extends App {
+          |  println(os.pwd)
+          |}
+          |""".stripMargin
+      )
+
+      val runProc = os.proc(TestUtil.cli, "--power", "compile", ".", extraOptions)
+        .call(cwd = root, stderr = os.Pipe)
+
+      expect(!runProc.err.trim.contains("Using directives detected in multiple files"))
+    }
+  }
+
+  test("fix with test scope") {
+    val mainSubPath = os.rel / "src" / "Main.scala"
+    val testSubPath = os.rel / "test" / "MyTests.scala"
+    val inputs = TestInputs(
+      mainSubPath ->
+        s"""//> using objectWrapper
+           |//> using dep "com.lihaoyi::os-lib:0.9.1"
+           |
+           |//> using test.dep "org.typelevel::cats-core:2.9.0"
+           |
+           |package com.foo.main
+           |
+           |object Main extends App {
+           |  println(os.pwd)
+           |}
+           |""".stripMargin,
+      testSubPath ->
+        s"""//> using options -Xasync, -Xfatal-warnings
+           |//> using dep org.scalameta::munit::0.7.29
+           |
+           |package com.foo.test.bar
+           |
+           |class MyTests extends munit.FunSuite {
+           |  test("bar") {
+           |    assert(2 + 2 == 4)
+           |    println("Hello from " + "tests")
+           |  }
+           |}
+           |""".stripMargin,
+      os.rel / projectFileName ->
+        s"""//> using lib com.lihaoyi::pprint:0.6.6
+           |""".stripMargin
+    )
+
+    inputs.fromRoot { root =>
+
+      val fixOutput = os.proc(TestUtil.cli, "--power", "fix", ".", "-v", "-v", extraOptions)
+        .call(cwd = root, mergeErrIntoOut = true).out.trim()
+
+      assertNoDiff(
+        fixOutput,
+        """Extracting directives from project.scala
+          |Extracting directives from src/Main.scala
+          |Removing directives from project.scala
+          |Removing directives from src/Main.scala
+          |Extracting directives from test/MyTests.scala
+          |Removing directives from test/MyTests.scala""".stripMargin
+      )
+
+      val projectFileContents = os.read(root / projectFileName)
+      val mainFileContents    = os.read(root / mainSubPath)
+      val testFileContents    = os.read(root / testSubPath)
+
+      assertNoDiff(
+        projectFileContents,
+        """// Main
+          |//> using objectWrapper
+          |//> using dependency "com.lihaoyi::os-lib:0.9.1", "com.lihaoyi::pprint:0.6.6"
```

Comment:
I think the `fix` command shouldn't generate directives that are separated by commas. It should enough to specify values without using commas.